third-party/realdds/src/dds-stream-server.cpp

```diff
@@ -117,6 +118,15 @@ void dds_stream_server::open( std::string const & topic_name, std::shared_ptr< d
     auto topic = topics::device::image::create_topic( publisher->get_participant(), topic_name.c_str() );
     _writer = std::make_shared< dds_topic_writer >( topic, publisher );
 
+
+    run_stream();
+}
+
+void dds_stream_server::run_stream()
```

Comment:
`run_stream()` was supposed to be a way to override how the stream is run (i.e., qos), not to configure callbacks on it. That's why the readers-changed callback was set inside open() and not here.
I think the proper way to do it is to have a virtual protected `create_topic()` call to handle different topic types, but leave it alone for now, we can change later.